PhoneDirectory.java

```diff
@@ -0,0 +1,46 @@
+import java.io.*;
+import java.util.*;
+
+class PhoneDirectory{
+
+    //Time Complexity : 0(1) we don't account for constructor as it is called only once
+    //Space Complexity: 0(n) where n is the max number
+    //Did it successfully run on leetcode: No, I don't have leetcode premium
+    //Did you face any problem while coding: No
+
+    //In brief explain your code
+
+    HashSet<Integer> set;   //we are using a hashset to query if a no is available or not
+    Queue<Integer> q;   //we are using a queue to store all the no's and also return the next available no.
+
+    public PhoneDirectory(int maxNumbers){
+        set = new HashSet<>();
+        q = new LinkedList<>();
+        for(int i = 0; i < maxNumbers; i++){
+            set.add(i); //storing all the available no's at first till max no's
+            q.add(i);
+        }
+    }
+
+    public int get(){
+        if(!q.isEmpty()){
```

Comment:
Here only you have to check for queue is isEMPTY(), But here you're returning -1 if q is not empty which is wrong, rest of this is fine.